mat64/dense.go

```diff
@@ -417,3 +420,25 @@ func (m *Dense) Augment(a, b Matrix) {
 	w := m.View(0, ac, br, bc).(*Dense)
 	w.Copy(b)
 }
+
+// MarshalBinary encodes the receiver into a binary form and returns the result.
+func (m Dense) MarshalBinary() ([]byte, error) {
+	buf := new(bytes.Buffer)
+	enc := binary.NewEncoder(buf)
```

Comment:
Using this package seems like a bad idea to me. It supports variable-length encoding, but does so by adding the size. Here's how I think it should work. Dense (etc) implement GobEncoder, which you use to get compact representations of your matrix, if, for example, your matrix is sparse or something. Then, it also implements MarshalBinary, where it first encodes all of size parameters (as int64 rather than int), and then encodes the floats at fixed-width (8 bytes). This does two things. First, it means that this is a portable binary representation with a very simple spec. There are three int64 (row, col, Stride), and then row*Stride + col  IEEE float64 numbers each at 8 bytes. Very easy to decode into other languages/programs. Secondly, because the sizes are known a-priori, MarshalBinary can be coded in parallel, which can have a big speedup for large matrices. 

In short: GobEncoder to have a compact representation, MarshalBinary to have parallel I/O and portable format.